store/sqlstore/reaction_store.go

```diff
@@ -228,6 +228,132 @@ func (s *SqlReactionStore) PermanentDeleteBatch(endTime int64, limit int64) (int
 	return rowsAffected, nil
 }
 
+func (s *SqlReactionStore) GetTopForTeamSince(teamID string, userID string, since int64, offset int, limit int) ([]*model.TopReactions, error) {
+	var reactions []*model.TopReactions
+
+	// Get reactions from all public channels within the team, and private channels that the user is a member of
+	query := `
```

Comment:
These queries still need to be optimized. Right now in Postgres it takes about 250ms for a user and 4s for the team (on the 15M post database over 28 days). I was thinking of just getting the endpoints setup so we can test and play around with them, and then optimize the queries (and performance elsewhere) once all the tests are in place so that they're easier to validate. I can create a ticket in JIRA for that as well.